src/osemgrep/networking/Rule_fetching.ml

```diff
@@ -114,7 +114,7 @@ let fetch_content_from_url_async ?(token_opt = None) (url : Uri.t) :
     let headers =
       match token_opt with
       | None -> None
-      | Some token -> Some [ ("authorization", "Bearer " ^ token) ]
+      | Some token -> Some [ ("Authorization", "Bearer " ^ token) ]
```

Comment:
Do we use caps elsewhere? If not I'd rather leave as-is. If we do then consistency is good.

Reply:
I'd prefer we follow the standard convention: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

Reply:
I don't think it is fair to characterize that as a standard convention on the basis of MDN's page names.

While HTTP 1 is (nominally) case-insensitive. HTTP/2 does not support uppercase names (https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2), and several services we interact with expect certain hashes or other values to be based on the lowercase versions (e.g., AWS: https://docs.aws.amazon.com/IAM/latest/UserGuide/create-signed-request.html#:~:text=Each%20header%20name,use%20lowercase%20characters.)

Reply:
For context, the MDN page linked references the [RFC 9110 spec](https://www.rfc-editor.org/rfc/rfc9110#field.authorization) which uses the HTTP/1.1 spec where as you noted header values should be treated as case-insensitive. Canonically, I've seen these header values typically capitalized e.g. basic test case for Flask framework: https://github.com/pallets/flask/blob/d61198941adcb191ddb591f08d7d912e40bde8bc/tests/test_basic.py#L1120


In the most recent [RFC 9113 ](https://www.rfc-editor.org/rfc/rfc9113.html#name-http-fields) you're correct that `Field names MUST be converted to lowercase when constructing an HTTP/2 message.` for HTTP/2

I'm not particular where we want to enforce lowercasing to support HTTP/2  and this requirement wasn't well known to me :)

Reply:
Great – same page here!